Automattic/newspack-newsletters
# Adds a basic Letterhead advertising integration 👋

### All Submissions:






### Changes proposed in this Pull Request:

Hey there 👋. This pull requests adds a very simple [Letterhead](https://tryletterhead.com) integration with Newspack, which allows folks to use Letterhead self-service ads in their newsletters. The only interface is a new setting where a user can optionally drop a Letterhead API key. After that, the implementation works the same as Newspack Newsletter's native ad post types, inserting on save.



Closes # .

### How to test the changes in this Pull Request:

1. Ping me for an API key : ) and an account. You'll be able to log in to your account by requesting a new password, once I've generated an admin account for ya.

2. Add the key to the Newsletters Settings field (at the bottom). This will connect you to a dummy account called Black Bitter Coffee I set up at https://app.tryletterhead.com 

3. You can check the promotions or create some more in the app, but all you should need to do in Newspack is send a test email for a day where there is an ad.

4. You should expect to see one or more promotions in your test email. They'd look like:

![image](https://user-images.githubusercontent.com/1489144/106050457-38030080-60b5-11eb-96c8-dced67c48499.png)


### Other information:

* [3] Have you successfully ran tests with your changes locally?

**Other changes**: I was invited to do a little cleanup here while in the plugin, so I did a little bit of abstraction - turning big functions into smaller functions. I tried to deeply comment everything to help devs navigate.



Edited 9/17 to update some URLs.
